torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -174,6 +174,53 @@ def affine_image_pil(
     return _FP.affine(img, matrix, interpolation=pil_modes_mapping[interpolation], fill=fill)
 
 
+def affine_bounding_box(
+    bounding_box: torch.Tensor,
+    format: features.BoundingBoxFormat,
+    image_size: Tuple[int, int],
+    angle: float,
+    translate: List[float],
+    scale: float,
+    shear: List[float],
+    center: Optional[List[float]] = None,
+) -> torch.Tensor:
+    original_shape = bounding_box.shape
+    bounding_box = convert_bounding_box_format(
+        bounding_box, old_format=format, new_format=features.BoundingBoxFormat.XYXY
+    ).view(-1, 4)
+
+    dtype = bounding_box.dtype if torch.is_floating_point(bounding_box) else torch.float32
+    device = bounding_box.device
+
+    if center is None:
+        height, width = image_size
+        center_f = [width * 0.5, height * 0.5]
+    else:
+        center_f = [float(c) for c in center]
+
+    translate_f = [float(t) for t in translate]
+    affine_matrix = torch.tensor(
+        _get_inverse_affine_matrix(center_f, angle, translate_f, scale, shear, inverted=False),
+        dtype=dtype,
+        device=device,
+    ).view(2, 3)
+    # bboxes to 4 points like:
+    # [(xmin, ymin, 1), (xmax, ymin, 1), (xmax, ymax, 1), (xmin, ymax, 1), ...]
+    points = bounding_box[:, [[0, 1], [2, 1], [2, 3], [0, 3]]].view(-1, 2)
+    points = torch.cat([points, torch.ones(points.shape[0], 1)], dim=-1)
+    transformed_points = points @ affine_matrix.T
+    # reshape transformed points to [N boxes, 4 points, x/y coords]
+    transformed_points = transformed_points.view(-1, 4, 2)
+    # compute bounding box from 4 transformed points:
+    out_bbox_mins, _ = torch.min(transformed_points, dim=1)
+    out_bbox_maxs, _ = torch.max(transformed_points, dim=1)
+    out_bboxes = torch.cat([out_bbox_mins, out_bbox_maxs], dim=1)
```

Comment:
Question: If I understand correctly, this converts from 4 points to the X1Y1, X2Y2 encoding of BBoxes. Right?

Reply:
Yes, if I understand correctly your question :) 
Lines 213-217 do the following. `transformed_points` are transformed 4 bbox points for N bboxes (N, 4, 2). 4 points correspond to transformed top-left, top-right, bottom-left and bottom-right bbox corners.
Next, we search for bounding box of those transformed 4 points and we obtain for each bbox min x,y and max x,y coordinates. Next we put them together to have [xmin, ymin, xmax, ymax] for each bbox.